xellgrid/grid.py

```diff
@@ -1156,9 +1165,20 @@ def _duplicate_last_row(self):
                            scroll_to_row=self._df.index.get_loc(max_index))
         return max_index + 1
 
-    def _add_empty_row(self):
-        self._df = pd.DataFrame([[np.nan] * len(self._df.columns)], columns=self._df.columns).concat(self._df,
-                                                                                                     ignore_index=True)
+    def _add_empty_row(self, row_index):
+        """Add empty row into current row_index
+
+        Parameters
+        ----------
+            row_index (int): current selected row index
+        """
+        new_row = self._df.iloc[[row_index]]
+        new_row = zero_out_dataframe(new_row)
```

Comment:
this is the best way i can think of, the idea is to add an "empty" row but don't change the original type.
for dtype like `category`, there won't be any default value but inherited the selected row.
so the idea is to duplicatd the selected row and zero out all possible dtypes. 
we can discuss this behavior further

Reply:
In Excel, a “Empty" role is a empty role. I think the best option is to make all values in the row as 
" ", this will provide the best user friendly experience.